File: Forge/src/main/java/hantonik/atomic/core/event/RegisterRecipesEvent.java
package hantonik.atomic.core.event;

import hantonik.atomic.core.utils.helpers.RecipeHelper;
import net.minecraft.world.item.crafting.Recipe;
import net.minecraft.world.item.crafting.RecipeManager;
import net.minecraftforge.eventbus.api.Event;

public class RegisterRecipesEvent extends Event {
    private final RecipeManager manager;

    public RegisterRecipesEvent(RecipeManager manager) {
        this.manager = manager;
    }

    public RecipeManager getRecipeManager() {
        return this.manager;
    }

    public void register(Recipe<?> recipe) {
        RecipeHelper.addRecipe(recipe);
    }
}

File: Forge/src/main/kotlin/hantonik/atomic/core/event/RegisterRecipesEvent.kt
package hantonik.atomic.core.event

import hantonik.atomic.core.util.helper.ACRecipeHelper
import net.minecraft.world.item.crafting.Recipe
import net.minecraft.world.item.crafting.RecipeManager
import net.minecraftforge.eventbus.api.Event

open class RegisterRecipesEvent(val recipeManager: RecipeManager) : Event() {
    open fun register(recipe: Recipe<*>) {
        ACRecipeHelper.addRecipe(recipe)
    }
}